tests/integration/security/reachability/mtls_test.go

```diff
@@ -0,0 +1,186 @@
+	appsInstance := apps.NewOrFail(ctx, t, apps.Config{Pilot: pilot})
+
+	aApp, _ := appsInstance.GetAppOrFail("a", t).(apps.KubeApp)
+	bApp, _ := appsInstance.GetAppOrFail("b", t).(apps.KubeApp)
+
+	headlessApp, _ := appsInstance.GetAppOrFail("headless", t).(apps.KubeApp)
+	// App without sidecar.
+	nakedApp, _ := appsInstance.GetAppOrFail("t", t).(apps.KubeApp)
+
+	testCases := []struct {
+		configFile  string
+		namespace   string
+		connections []connect.Connection
+	}{
+		{
+			configFile: "global-mtls-on.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            aApp,
+					To:              headlessApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-permissive.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-off.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "single-port-mtls-on.yaml",
+			namespace:  appsInstance.Namespace().Name(),
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+	}
+
+	for i, c := range testCases {
+		t.Run(fmt.Sprintf("%v - %v", i, c.configFile), func(t *testing.T) {
```

Comment:
nit: you may want to try running the test in GoLand to see what the test names look like before submitting.